internal/pkg/views/notifications.go

```diff
@@ -0,0 +1,175 @@
+	if url == "" {
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			Failure: true,
+			Message: "Item `" + display + "` doesn't support delete",
+			Timeout: time.Second * 5,
+		})
+		return
+	}
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		for _, i := range pendingDeletes {
+			if i.url == url {
+				eventing.SendStatusEvent(eventing.StatusEvent{
+					Failure: true,
+					Message: "Item already `" + display + "` in pending delete list",
+					Timeout: time.Second * 5,
+				})
+				return nil
+			}
+		}
+
+		pendingDeletes = append(pendingDeletes, pendingDelete{
+			url:     url,
+			display: display,
+		})
+
+		return nil
+	})
+}
+
+// ConfirmDelete delete all queued/pending deletes
+func ConfirmDelete() {
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		pending := pendingDeletes
+		event, _ := eventing.SendStatusEvent(eventing.StatusEvent{
+			InProgress: true,
+			Message:    "Starting to delete items",
+			Timeout:    time.Second * 15,
+		})
+
+		for _, i := range pending {
+			_, err := armclient.DoRequest(context.Background(), "DELETE", i.url)
+			if err != nil {
+				event.Failure = true
+				event.Message = "Failed to delete `" + i.display + "` with error:" + err.Error()
+				event.Update()
+
+				// In the event that a delete fails in the
+				// batch of pending deletes lets give up on the rest
+				// as something might have gone wrong and best
+				// to be cautious
+				return nil
+			}
+
+			event.Message = "Deleted: " + i.display
+			event.Update()
+		}
+
+		event.Message = "Delete request sent"
+		event.InProgress = false
+		event.Update()
+
+		pendingDeletes = []pendingDelete{}
+
+		return nil
+	})
+}
+
+// ClearPendingDeletes removes all pending deletes
+func ClearPendingDeletes() {
+	deleteMutex.Lock()
+	gui.Update(func(g *gocui.Gui) error {
+
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			InProgress: true,
+			Message:    "Clearing pending deletes",
+			Timeout:    time.Second * 2,
+		})
+
+		pendingDeletes = []pendingDelete{}
+		deleteMutex.Unlock()
+
+		return nil
+	})
+}
+
+// NotificationWidget controls the statusbar
+type NotificationWidget struct {
+	ConfirmDeleteKeyBinding       string
+	ClearPendingDeletesKeyBinding string
+	name                          string
+	x, y                          int
+	w                             int
+}
+
+// NewNotificationWidget create new instance and start go routine for spinner
+func NewNotificationWidget(x, y, w int, hideGuids bool, g *gocui.Gui) *NotificationWidget {
+	widget := &NotificationWidget{
+		name: "notificationWidget",
+		x:    x,
+		y:    y,
+		w:    w,
+	}
+	gui = g
+	return widget
+}
+
+// Layout draws the widget in the gocui view
+func (w *NotificationWidget) Layout(g *gocui.Gui) error {
+	// Don't draw anything if no pending deletes
+	if len(pendingDeletes) < 1 {
+		g.DeleteView(w.name)
+		return nil
+	}
+
+	height := len(pendingDeletes)*1 + 7
```

Comment:
Does this need to be capped with a scroll or something?

Reply:
Another awesome spot. Thinking about it I'm not keep on scrolling as could easily miss items that are at the top of the list and think they're not on delete list. What do you think of something like this inside `AddPendingDelete`?

```golang

	// Don't add more items than we can draw on the
	// current terminal size
	_, yMax := gui.Size()
	if len(pendingDeletes) > (yMax - 12) {
		eventing.SendStatusEvent(eventing.StatusEvent{
			Failure: true,
			Message: "Can't add `" + display + "` run out of space!",
			Timeout: time.Second * 5,
		})
		return
	}
```

Reply:
Seems fair enough yh